### PR TITLE
Add polyfill for pre-ES6 form of `Function.name`

### LINKED ISSUE #3: Submit to polyfill.io
Hi @JamesMGreene,

have you considered to submit this Polyfill to https://github.com/Financial-Times/polyfill-service?
I'd love to see it available there!

Best,
Chris

### LINKED ISSUE #943: Support for Function.name
Haven't seen this on the feature list and no issue or pull-request either.

I'd like to see [Function.name](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name) supported by Polyfill.io.

### PR DESCRIPTION
**Pull request by JamesMGreene**
Fixes #943 
Closes JamesMGreene/Function.name#3

### CODE REVIEW

**Review by triblondon — COMMENTED** [review]
Hi @JamesMGreene, thanks for this.  Could you check off all the points in this list:

https://polyfill.io/v2/docs/contributing/common-scenarios#adding-browsers

To help us merge this ASAP?

**Comment by triblondon** [comment]
Pushed to a local branch for CI

File: polyfills/Function/name/tests.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+it('returns the name of named function expressions', function () {
+	var foo = function bar() {};
+
+	proclaim.equal(foo.name, 'bar');
+});
+
+it('returns an empty string for anonymous function expressions', function () {
+	var anon = function () {};
+	proclaim.equal(anon.name, '');
```
Comment (falsandtru) [inline polyfills/Function/name/tests.js]: In latest chrome, `anon.name` is `'anon'`, not an empty string.

**Review by falsandtru — CHANGES_REQUESTED** [review]

File: polyfills/Function/name/tests.js (hunk 1 above)
Comment (triblondon) [reply polyfills/Function/name/tests.js]: Agreed:

image

**Comment by falsandtru** [comment]
@triblondon As I commented, some tests fail on the latest browsers. So we need to run tests on them. Can you add the target browsers such as Chrome latest(53+), Chrome canary, and Android 5+?

**Comment by triblondon** [comment]
@falsandtru https://github.com/Financial-Times/polyfill-service/pull/954

**Comment by JamesMGreene** [comment]
> Check reliable sources of browser support information, e.g. MDN, caniuse and Kangax. Tell us what these sources say about the browser/polyfill combo that you want to add or remove, and link to the relevant pages from your PR.
- [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name)
- [MDN - Browser Compatibility](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name#Browser_compatibility)
- [caniuse](http://kangax.github.io/compat-table/es6/#test-function_name_property) (for the enhanced ES6 form, though)

**Basic support**
- Chrome 33.0
- Firefox (Yes)
- IE 12
- Opera (Yes)
- Safari (Yes)

> Explain how you have tested support (or lack of) in the browser concerned. Ideally, include a screenshot of the relevant browser running our test suite for that feature.

Ran the newly added tests against my own SauceLabs account with `grunt ci`.

> State your agreement to our contribution terms.

I agree to the contribution terms.

**Comment by JamesMGreene** [comment]
@triblondon FYI, I rebased your changes from `master` into the PR branch now, so the merge commit no longer shows up.

**Review by triblondon — APPROVED** [review]
Outstanding.

**Comment by triblondon** [comment]
Pushed to local branch for CI.  Note to self: because it was rebased I had to force push.

**Comment by JamesMGreene** [comment]
:+1: 💝

**Comment by JakeChampion** [comment]
Thanks for the contribution @JamesMGreene! Did you know we offer one month's free premium access to all FT content (normally $50) to anyone who has a pull request merged in this repository? If you want to claim it, there's [one Google form](https://docs.google.com/a/ft.com/forms/d/1aykPiYBwqKpKY6nvGVi43wOU_5epr3XLgE00Vi52SMQ/viewform) to fill out :-)

**Pull request merged**